standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/properties/HMSPropertyStoreRemoteTest.java

```diff
@@ -0,0 +1,233 @@
+import static org.apache.hadoop.hive.metastore.properties.PropertyType.DATETIME;
+import static org.apache.hadoop.hive.metastore.properties.PropertyType.DOUBLE;
+import static org.apache.hadoop.hive.metastore.properties.PropertyType.INTEGER;
+import static org.apache.hadoop.hive.metastore.properties.PropertyType.JSON;
+import static org.apache.hadoop.hive.metastore.properties.PropertyType.STRING;
+
+public class HMSPropertyStoreRemoteTest extends HMSPropertyStoreTest {
+  //private static final String NS = "hms";
+  protected HiveMetaStoreClient client;
+  private boolean isServerStarted = false;
+  protected int port;
+
+  boolean createStore(Configuration conf, Warehouse wh) {
+    try {
+      MetaStoreTestUtils.setConfForStandloneMode(conf);
+      objectStore = new ObjectStore();
+      objectStore.setConf(conf);
+      //TestObjectStore.dropAllStoreObjects(objectStore);
+      HMSHandler.createDefaultCatalog(objectStore, wh);
+      // configure object store
+      objectStore.createDatabase(new DatabaseBuilder()
+          .setCatalogName("hive")
+          .setName(DB1)
+          .setDescription("description")
+          .setLocation("locationurl")
+          .build(conf));
+    } catch(InvalidObjectException | MetaException | InvalidOperationException xmeta) {
+      throw new PropertyException("unable to initialize server", xmeta);
+    }
+    return true;
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    conf = MetastoreConf.newMetastoreConf();
+    MetastoreConf.setBoolVar(conf, MetastoreConf.ConfVars.HIVE_IN_TEST, true);
+    // Events that get cleaned happen in batches of 1 to exercise batching code
+    MetastoreConf.setLongVar(conf, MetastoreConf.ConfVars.EVENT_CLEAN_MAX_EVENTS, 1L);
+    MetaStoreTestUtils.setConfForStandloneMode(conf);
+    if (isServerStarted) {
+      Assert.assertNotNull("Unable to connect to the MetaStore server", client);
+      return;
+    }
+    port = MetaStoreTestUtils.startMetaStoreWithRetry(HadoopThriftAuthBridge.getBridge(), conf);
+    System.out.println("Starting MetaStore Server on port " + port);
+    isServerStarted = true;
+
+    Warehouse wh = new Warehouse(conf);
+    boolean inited = createStore(conf, wh);
+    LOG.info("MetaStore Thrift Server test initialization " + (inited? "successful":"failed"));
+    // This is default case with setugi off for both client and server
+    client = createClient();
+  }
+
+
+  protected HiveMetaStoreClient createClient() throws Exception {
+    MetastoreConf.setVar(conf, MetastoreConf.ConfVars.THRIFT_URIS, "thrift://localhost:" + port);
+    MetastoreConf.setBoolVar(conf, MetastoreConf.ConfVars.EXECUTE_SET_UGI, false);
+    HiveMetaStoreClient client = new HiveMetaStoreClient(conf);
+    return client;
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    MetaStoreTestUtils.close(port);
+    super.tearDown();
+  }
+
+  @Test
+  public void testHMSProperties() throws Exception {
```

Comment:
I think we can just setup the config to use remote HMS and then run the tests from HMSPropertyStoreTest class. We currently do that for few HMS unit tests.

Reply:
Refactored the unit tests (and added a Servlet skeleton to test another transport).